SimpleNotePad.java

```diff
@@ -0,0 +1,225 @@
+            pjob.setJobName("Sample Command Pattern");
+            pjob.setCopies(1);
+            pjob.setPrintable(new Printable() {
+                public int print(Graphics pg, PageFormat pf, int pageNum) {
+                    if (pageNum>0)
+                        return Printable.NO_SUCH_PAGE;
+                    pg.drawString(d.getText(), 500, 500);
+                    paint(pg);
+                    return Printable.PAGE_EXISTS;
+                }
+            });
+
+            if (pjob.printDialog() == false)
+                return;
+            pjob.print();
+        } catch (PrinterException pe) {
+            JOptionPane.showMessageDialog(null,
+                    "Printer error" + pe, "Printing error",
+                    JOptionPane.ERROR_MESSAGE);
+        }
+    }
+    
+    private void saveFunction() {
+    	File fileToWrite = null;
+        JFileChooser fc = new JFileChooser();
+        int returnVal = fc.showSaveDialog(null);
+        if (returnVal == JFileChooser.APPROVE_OPTION) {
+        	fileToWrite = fc.getSelectedFile();
+            try {
+                PrintWriter out = new PrintWriter(new FileWriter(fileToWrite));
+                out.println(d.getText());
+                JOptionPane.showMessageDialog(null, "File is saved successfully...");
+                out.close();
+            } catch (IOException ex) {
+            }
+        }
+    }
+    
+    public static void main(String[] args) {
+        SimpleNotePad app = new SimpleNotePad();
+    }
+    
+    @Override
+    public void actionPerformed(ActionEvent e) {
+    	String actionCommand = e.getActionCommand();
+        if (actionCommand.equals("open")) {
+        	JFileChooser fileChooser = new JFileChooser();
+        	int returnVal = fileChooser.showOpenDialog(null);
+        	if (returnVal == JFileChooser.APPROVE_OPTION) {
+        		openFile(fileChooser.getSelectedFile());
+        	}
+        } else if (actionCommand.equals("new")) {
+            d.setText("");
+        } else if (actionCommand.equals("save")) {
+            saveFunction();
+        } else if (actionCommand.equals("print")) {
+            printFunction();
+        } else if (actionCommand.equals("recent0")) {
+        	openFile(recentFile[0]);
+        } else if (actionCommand.equals("recent1")) {
+        	openFile(recentFile[1]);
+    	} else if (actionCommand.equals("recent2")) {
+        	openFile(recentFile[2]);
+    	} else if (actionCommand.equals("recent3")) {
+        	openFile(recentFile[3]);
+    	} else if (actionCommand.equals("recent4")) {
+        	openFile(recentFile[4]);
+    	} else if (actionCommand.equals("copy")) {
+            d.copy();
+        } else if (actionCommand.equals("paste")) {
+            d.paste();
+        } else if (actionCommand.equals("simple_replace")) {
+        	String replace = JOptionPane.showInputDialog(null, "Replace or insert with:", "Input", JOptionPane.PLAIN_MESSAGE);
+        	d.replaceSelection(replace);

```

Comment:
```suggestion
        	if (replace != null) {
        	    d.replaceSelection(replace);
        	}
```
Need to make sure replace is not null or whatever is in the clipboard will be inserted.